CRM/Contribute/BAO/Contribution.php

```diff
@@ -456,17 +456,22 @@ public static function getPaymentProcessorReadyAddressParams($params, $billingLo
 
   /**
    * Get the number of terms for this contribution for a given membership type
-   * based on querying the line item table and relevant price field values
+   * based on querying the line item table and relevant price field values.
    * Note that any one contribution should only be able to have one line item relating to a particular membership
-   * type
+   * type.
    *
-   * @param int $membershipTypeID
+   * For offline recurring Contribution we return '0' to not extend Membership.
```

Comment:
I presume this is against core mostly for the purpose of seeing how tests run - but I would think the 'reason' not to extend the membership would be that the invoice would still be 'pending' - this seems like it belongs in an extension because it seems quite tied to a use case